private/buf/bufsync/bufsync.go

```diff
@@ -270,6 +280,26 @@ type ModuleDefaultBranchGetter func(
 	module bufmoduleref.ModuleIdentity,
 ) (string, error)
 
+// OldTagsAttacher is invoked when a commit with valid modules is found close past the start sync
+// point for such module. The Syncer assumes that the "old" commit is already synced, so it will
+// attempt to attach existing tags using that git hash, in case they were recenly created or moved
+// there.
+//
+// A common scenario is SemVer releases: a commit is pushed to the default Git branch, the sync
+// process triggers and completes, and some minutes later that commit is tagged "v1.2.3". The next
+// time the sync command runs, this attacher would pick such tag and attach it to the correct BSR
+// commit.
+//
+// It's expected to return the BSR commit name to which the tags were attached.
```

Comment:
Nit: instead of `OldTagsAttacher`, can we call this something like `BackfillTagsAttacher`?

```suggestion
// OldTagsAttacher is invoked when a commit with valid modules is found within a lookback threshold past the start sync
// point for such module. The Syncer assumes that the "old" commit is already synced, so it will
// attempt to attach existing tags using that git hash, in case they were recently created or moved
// there.
//
// A common scenario is SemVer releases: a commit is pushed to the default Git branch, the sync
// process triggers and completes, and some minutes later that commit is tagged "v1.2.3". The next
// time the sync command runs, this attacher would pick such tag and attach it to the correct BSR
// commit.
//
// It's expected to return the BSR commit name to which the tags were attached.
```